src/AutoRest.CSharp/Mgmt/Generation/ResourceContainerWriter.cs

```diff
@@ -396,16 +396,6 @@ public ParameterMapping(Parameter parameter, bool isPassThru, string valueExpres
             }
         }
 
-        /// <summary>
-        /// Is the input type string or an Enum that is modeled as string.
-        /// </summary>
-        /// <param name="type">Type to check.</param>
-        /// <returns>Is the input type string or an Enum that is modeled as string.</returns>
-        private bool IsStringLike(CSharp.Generation.Types.CSharpType type)
-        {
-            return type.Equals(typeof(string)) || type.Implementation is EnumType enumType && enumType.BaseType.Equals(typeof(string));
```

Comment:
root cause is type doesn't always have implementation, it could be some other system type